lib/application.js

```diff
@@ -592,7 +593,14 @@ app.render = function(name, options, fn){
 
 app.listen = function(){
   var server = http.createServer(this);
-  return server.listen.apply(server, arguments);
+  var args = Array.prototype.slice.call(arguments);
+  if (typeof args[args.length - 1] === 'function'){
+    var done = Array.prototype.pop.call(args);
+    eeFirst([[server, 'error', 'listening']], function(err, ee, event, eventArgs){
+      done.apply(this, eventArgs);
```

Comment:
Should probably use the same `context` as the listening event does: against `ee` instead of `app`.